File: javascript/fibonacci.test.js
const fibonacci = require('./fibonacci')

test('fib 1', () => {
  let fib = fibonacci()
  expect(fib.calc(1)).toBe(1)
})

test('fib 2', () => {
  let fib = fibonacci()
  expect(fib.calc(2)).toBe(1)
})

test('fib 3', () => {
  let fib = fibonacci()
  expect(fib.calc(3)).toBe(2)
})

test('fib 4', () => {
  let fib = fibonacci()
  expect(fib.calc(4)).toBe(3)
})

test('fib 5', () => {
  let fib = fibonacci()
  expect(fib.calc(5)).toBe(5)
})

test('fib 6', () => {
  let fib = fibonacci()
  expect(fib.calc(6)).toBe(8)
})

test.skip('fib 90', () => {
  let fib = fibonacci()
  expect(fib.calc(90)).toBe(2880067194370816120)
})

File: java/test/FibonacciTest.java
import org.junit.Before;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

public class FibonacciTest {

  private Fibonacci fibonacci;

  @Before
  public void setup() {
    fibonacci = new Fibonacci();
  }

  @Test
  public void fib1() {
    assertEquals(1, fibonacci.calc(1));
  }

  @Test
  public void fib2() {
    assertEquals(1, fibonacci.calc(2));
  }

  @Test
  public void fib3() {
    assertEquals(2, fibonacci.calc(3));
  }

  @Test
  public void fib4() {
    assertEquals(3, fibonacci.calc(4));
  }

  @Test
  public void fib5() {
    assertEquals(5, fibonacci.calc(5));
  }

  @Test
  public void fib6() {
    assertEquals(8, fibonacci.calc(6));
  }

  @Test
  public void fib90() {
    assertEquals(2880067194370816120L, fibonacci.calc(90));
  }
}
